models/user.go

```diff
@@ -1360,6 +1360,18 @@ func SearchUsers(opts *SearchUserOptions) (users []*User, _ int64, _ error) {
 		Find(&users)
 }
 
+//SearchUsersAPI takes options i.e keyword and part of user name to search
+//it returns all the results found and the number of total results.
+func SearchUsersAPI(opts *SearchUserOptions) ([]*User, int64, error) {
```

Comment:
I would prefer if instead of new function `SearchUsers` would be reused and changed that `PageSize: -1`  would mean all.

For example something like this:

```go
// SearchUsers takes options i.e. keyword and part of user name to search,
// it returns results in given range and number of total results.
func SearchUsers(opts *SearchUserOptions) (users []*User, _ int64, _ error) {
	cond := opts.toConds()
	count, err := x.Where(cond).Count(new(User))
	if err != nil {
		return nil, 0, fmt.Errorf("Count: %v", err)
	}

	if opts.PageSize == 0 || opts.PageSize > setting.UI.ExplorePagingNum {
		opts.PageSize = setting.UI.ExplorePagingNum
	}
	if opts.Page <= 0 {
		opts.Page = 1
	}
	if len(opts.OrderBy) == 0 {
		opts.OrderBy = SearchOrderByAlphabetically
	}

	sess := x.Where(cond)
	if opts.PageSize > 0 {
		sess = sess.Limit(opts.PageSize, (opts.Page-1)*opts.PageSize)
	}

	users = make([]*User, 0, opts.PageSize)
	return users, count, sess.OrderBy(opts.OrderBy.String()).Find(&users)
}
```

Reply:
@lafriks I've just pushed these changes